Branch main,

DubovikovArtem opened the pull request:
Homework for lesson 18:

 create new package: lesson18
 create new class: BaseTest
 create new class: TestTask1
 create new class: TestCheckGps
 create new class: TestCheckLogin

 create file chromedriver.exe
 update configuration in pom.xml

DubovikovArtem committed:
Branch main,
Homework for lesson 18:

 create new package: lesson18
 create new class: BaseTest
 create new class: TestTask1
 create new class: TestCheckGps
 create new class: TestCheckLogin

 create file chromedriver.exe
 update configuration in pom.xml

apollyonix reviewed (changes requested):
А где, собственно, сами тестовые классы?)

apollyonix commented on com.hillel/src/test/java/lesson18/BaseTest.java:
```diff
@@ -0,0 +1,22 @@
+
+    @BeforeClass
+    public void tes() {
+        WebDriverManager.chromedriver().setup();
```
Поскольку ты используешь WebDriverManager, то уже нет необходимости таскать за собой в ресурсах бинарный хром драйвер, менеджер его сам скачает по необходимости

DubovikovArtem committed:
Branch main,
Homework for lesson 18:

 create new package: lesson18
 create new class: BaseTest
 create new class: TestTask1
 create new class: TestCheckGps
 create new class: TestCheckLogin

apollyonix commented on com.hillel/src/test/java/lesson18/TestTask1.java:
```diff
@@ -0,0 +1,34 @@
+            trs.findElements(By.tagName("td"));
+            listTd = trs.findElements(By.tagName("td"));
+            for (int j = 0; j < listTd.size(); j++) {
+                WebElement x = listTd.get(j);
```
тут можно сделать всё намного проще, сразу в локаторе указать, что нам нужны все 4-ые ячейки в каждой строке. и потом один раз пройтись по полученному списку.
в данной реализации - ты дёргаешь вебдрайвер, чтобы достать строку из таблицы, далее ты дёргаешь его для каждой ячейки (td) чтобы достать текст, и только после этого сравниваешь индекс с нужным, чтобы вывести на экран. каждое такое "дёрганье" довольно трудозатратно.

apollyonix commented on com.hillel/src/test/java/lesson18/TestCheckLogin.java:
```diff
@@ -0,0 +1,61 @@
+        String validResult = "You logged into a secure area!";
+        driver.get("https://the-internet.herokuapp.com/login");
+
+        driver.findElement(By.cssSelector("[type='text']")).sendKeys(validLogin);
```
тут ты уже столкнулся с пролемой, что в 3х разных тестах прихоидтся работать с одними и теми же элементами, 3 раза указывать одинаковые локаторы. дальше по курсу рассмотрим, как это можно упростить, чтобы в случае необходимости не надо было менять эти локаторы в кучу мест, а только в одном. так мы избавимся от дублирования кода

apollyonix reviewed (approved):
Всё выполнено верно
